### PR TITLE
[KMCompiler][NVIDIA]Add conj operator with view semantics

### PR DESCRIPTION
**Pull request by tspyc072**
## Summary

Adds the `conj` operator to FlagGems. For real (non-complex) tensors: returns the input tensor itself (identity, no memory allocation). For complex tensors: returns a view with the conjugate flag set via `torch.Tensor._conj()` (O(1) operation, no Triton kernel).

## Implementation

- **Core Logic**: Implemented in `src/flag_gems/ops/conj.py`. For real tensors, returns input directly. For complex tensors, calls `input._conj()` to return a view sharing the same underlying storage.
- **View Semantics**: Unlike physical computation, this implementation leverages PyTorch's native conjugate flag mechanism. No memory allocation, no Triton kernel launch, no kernel tuning required.
- **Registration**: Registered in `src/flag_gems/ops/__init__.py` and `src/flag_gems/__init__.py` (`_FULL_CONFIG`) to intercept `aten::conj` calls. Also registered in `conf/operators.yaml`.

## Testing

- **Accuracy**: `test_conj_complex` verifies `is_conj()` flag, `data_ptr()` sharing, and bitwise-identical results to native PyTorch reference. `test_conj_real` verifies identity return.
- **Coverage**: 30 test cases covering multiple shapes and dtypes (`complex64`, `complex128`, `float16`, `float32`, `bfloat16`).
- **Platform Compatibility**: Tests pass on **NVIDIA H20** (30/30).

## Performance (NVIDIA - H20)

`torch.conj` native implementation returns a view (zero-copy metadata operation). Our implementation also returns a view via `input._conj()`, so latency is identical to PyTorch native.

### NVIDIA conj Results (All Dtypes)

| Status | Torch Latency (ms) | Gems Latency (ms) | Gems Speedup | Size Detail |
|--------|-------------------|-------------------|--------------|-------------|
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1073741824])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([64, 64])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([4096, 4096])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([64, 512, 512])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1024, 1024, 1024])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1024, 1])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1024, 16])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1024, 256])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1024, 4096])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([1024, 65536])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([64, 64, 1])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([64, 64, 16])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([64, 64, 256])] |
| SUCCESS | 0.003 | 0.003 | **1.000** | [torch.Size([64, 64, 4096])] |

*All dtypes (complex64, float16, float32, bfloat16) achieve speedup=1.0 because both native PyTorch and Gems use view semantics.*

## Files Changed

- `src/flag_gems/ops/conj.py`: Core view-based implementation.
- `src/flag_gems/ops/__init__.py`: Exported `conj`.
- `src/flag_gems/__init__.py`: Registered `conj` in `_FULL_CONFIG`.
- `conf/operators.yaml`: Registered `conj` operator.
- `tests/test_conj.py`: Accuracy tests.
- `benchmark/test_conj.py`: Performance benchmark.

### CODE REVIEW

File: src/flag_gems/ops/__init__.py [hunk 1]
```diff
@@ -693,6 +694,7 @@
     "concat",
     "concatenate",
     "conj_physical",
+    "conj",
```
Comment (tengqm) [inline src/flag_gems/ops/__init__.py:662]: move this before line 696

File: src/flag_gems/ops/conj.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import logging
+import torch
+
+logger = logging.getLogger(__name__)
+
+def conj(input: torch.Tensor) -> torch.Tensor:
```
Comment (tengqm) [inline src/flag_gems/ops/conj.py:8]: pls register the operator into `/conf/operators.yaml`.

File: src/flag_gems/ops/conj.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    resolve_conj), the imaginary part is negated lazily.
+
+    This matches the semantics of torch.conj exactly: zero memory copy,
+    output and input share the same data pointer.
+    """
+    logger.debug("GEMS CONJ")
+    if not input.is_complex():
+        return input
+    # _conj() is a low-level TensorImpl method that toggles the conjugate flag
+    return input._conj()
```
Comment (tengqm) [inline src/flag_gems/ops/conj.py:17]: so ... this is not a Triton operator?

File: src/flag_gems/ops/__init__.py (hunk 1 above)
Comment (tspyc072) [reply src/flag_gems/ops/__init__.py:662]: Fixed. `conj` has been moved before `conj_physical` in `__all__`.

File: src/flag_gems/ops/conj.py (hunk 2 above)
Comment (tspyc072) [reply src/flag_gems/ops/conj.py:8]: Done. Registered `conj` in `conf/operators.yaml`.

File: src/flag_gems/ops/conj.py (hunk 3 above)
Comment (tspyc072) [reply src/flag_gems/ops/conj.py:17]: `conj` is a metadata operation (zero-copy view) by design, matching `torch.conj` semantics. We also provide a physical Triton implementation in `_nvidia/ops/conj.py` for backend completeness.

File: conf/operators.yaml [hunk 4]
```diff
@@ -1682,6 +1530,18 @@ ops:
...
+      Returns a view of the input tensor with the conjugate flag set.
+      For complex tensors, this is a zero-copy view sharing the same storage.
+    for:
+      - conj
+    labels:
+      - aten
+    kind:
+      - Tensor
+    stages:
+      - stable: '2.2'
```
Comment (tengqm) [inline conf/operators.yaml:1544]: ？

File: src/flag_gems/runtime/backend/_ascend/ops/conj.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import logging
```
Comment (tengqm) [inline src/flag_gems/runtime/backend/_ascend/ops/conj.py]: why we cannot use the generic implementation?

**Comment by bin913** [comment]
@tspyc072 Please fix failing check

**Comment by Caeruleann** [comment]
Please fix the failing checks, rebase your branch, and address the review comments above.
